### DISCUSSION TITLE
[Design] Website overhaul for skywalking.apache.org

### DISCUSSION THREAD

**Discussion by rickyzhangca** [apache/skywalking#7722] [Ideas]
I'm a summer student working on an improved design for [skywalking.apache.org](https://skywalking.apache.org/). After 2.5 months of designing, I'm showcasing some of the ideas here and sharing [an interactive prototype](https://www.figma.com/proto/y9aSNZD0I3PTo6oko8Z3gH/4-%E8%AE%BE%E8%AE%A1%E5%8E%9F%E5%9E%8B?node-id=10%3A3&viewport=241%2C48%2C0.27&scaling=min-zoom&starting-point-node-id=10%3A3&hide-ui=1) to hear feedbacks and thoughts from everyone. Any input would be appreciated. I'm happy to explain the design decisions behind any change.

Credits to @wu-sheng for being my mentor.

### Improved readability 

Several changes are introduced aiming to improve the readability. They include decluttering, restyling and reorganizing contents.

![image](https://user-images.githubusercontent.com/16908811/133546644-38994f9b-dbad-496f-842c-15f7f0ccc5ae.png)
![image](https://user-images.githubusercontent.com/16908811/133546775-ae0aeb04-0715-4653-9b15-e9dd79638d7e.png)

### More intuitive interactions

Some of the interactions are reworked to better involve user at where we should. For example, users can peek SkyWalking UI right at the landing scene, and find the `try live demo` button soon after. The logo wall simply self-scroll. 

![image](https://user-images.githubusercontent.com/16908811/133547038-37d057df-5c18-42e8-a19f-b76535758600.png)
![image](https://user-images.githubusercontent.com/16908811/133546924-b2880d26-d1f2-4715-bc6c-e69dfa3c1d34.png)

### Refined visual design

All designers love over-spend time on pixel-perfecting. Other than aesthetics, there are other good reasons behind. For example, a set of clean icons here lets you locate repos easier. 

![image](https://user-images.githubusercontent.com/16908811/133545940-47e58e94-ced3-41ae-b1bf-945eb44d5144.png)

### Check out the prototype!

"A prototype is worth a thousand pictures": [start](https://www.figma.com/proto/y9aSNZD0I3PTo6oko8Z3gH/4-%E8%AE%BE%E8%AE%A1%E5%8E%9F%E5%9E%8B?node-id=10%3A3&viewport=241%2C48%2C0.27&scaling=min-zoom&starting-point-node-id=10%3A3&hide-ui=1)

https://user-images.githubusercontent.com/16908811/133545759-cacd78fe-0354-45c8-84f2-ceeda214841b.mov

**Comment by wu-sheng** [comment]
**User** page seems a little small, I think we should use vertical space more.

**Comment by hanahmily** [comment]
Awesome. The website is more attractive than before.

All changes look brilliant to me except for the feature gallery
![WX20210916-145405](https://user-images.githubusercontent.com/1960537/133564487-c65a523b-e89c-4b51-8523-882807506b7f.png)

Could we place some diagrams or artworks instead of raw YAMLs here? The visitor has to zoom into every piece to find out the details of each one.

**Reply by rickyzhangca** [reply]
That is almost the reasoning behind this design, users can zoom into the technical details a bit more. I assumed that most of the users are comfortable reading the snippets.

**Comment by kezhenxu94** [comment]
The new design looks fantastic!

One thing to note, as per the Apache policy, we should not display the users in our home page, we have fixed this days ago and it seems to be on your new design still, you might need to adjust the home page of your new design accordingly.

**Reply by rickyzhangca** [reply]
Thank you! I noticed the [change](https://github.com/apache/skywalking-website/commit/e481896aff488819be6043663dd3e919b25c18be) but I thought it was to clean up the page. My bad, should've checked the commit! I'm curious to learn more about the policy to see to what scale we should not display them (e.g. we can't mention them in any way, or should not display trademarks, etc.) Could you please send me a link to it? Thanks. Wasn't able to find the exact place myself.

**Reply by kezhenxu94** [reply]
Hi @rickyzhangca, here is the guideline https://apache.org/foundation/marks/linking#projectthanks

The basic idea is not to let users think the project belongs to, or backed by any vendor or company, etc. ASF projects should be totally vendor-neutral, displaying the company logos in the home page might be misleading.

**Reply by kezhenxu94** [reply]
But, we can display them in secondary pages (menu), like the current location of our website.

**Reply by rickyzhangca** [reply]
Thanks, and I appreciate Apache for providing the guideline and rational behind it. Removing the logo wall. I wonder if it's ok to show usage data (etc. X tracing/day at Y, proud to empower Z companies)? These demonstrate the scale SkyWalking is used and capable of. The guideline didn't seem to have a clear cut here and this usage seems to comply the guideline. Correct me if wrong @kezhenxu94 @wu-sheng

**Reply by kezhenxu94** [reply]
Showing the usage data is feasible to me, as long as we don't show bias to any vendor or company, "x traces/day" is an objective statement and IMO it is ok to be on the website but "proud to empower y company" seems to be kind of subjective and might imply bias in some linguistic context.

**Reply by rickyzhangca** [reply]
edited, what do you think?
![image](https://user-images.githubusercontent.com/16908811/133942449-b4e4d9a9-e279-4313-9a8a-cde96846c1a8.png)

**Reply by kezhenxu94** [reply]
> edited, what do you think?
> 
> ![image](https://user-images.githubusercontent.com/16908811/133942449-b4e4d9a9-e279-4313-9a8a-cde96846c1a8.png)
> 
> 

Hi @rickyzhangca, sorry I didn't make it clear enough, "powered by" companies' names and/or logos are all not allowed in home page, the stats without company names are allowed. As per this policy, I think we have to put these in a secondary page and only keep a hyperlink in the homepage

**Reply by rickyzhangca** [reply]
removed, I liked this section

**Comment by heyanlong** [comment]
Stunning design!🥳

But I can't visually see the meaning of the colored buttons on the right.
![image](https://user-images.githubusercontent.com/3367958/133577587-4ab63e3f-cc4d-4dc4-9c3f-5cdbf43b643f.png)

**Reply by wu-sheng** [reply]
It means(from the original graph) what parts of observability data could be collected.

https://camo.githubusercontent.com/143779cb51ec9557528e9059d1386d6cbc905fb46c8c20603f2f4dc0fb2b8ab1/68747470733a2f2f736b7977616c6b696e672e6170616368652e6f72672f696d616765732f536b7957616c6b696e675f4172636869746563747572655f32303231303432342e706e673f743d3230323130343234

@rickyzhangca We should make sure it is clear.

**Reply by rickyzhangca** [reply]
It took me a long time to think about this and to be honest I don haven't the best solution yet. I'll chat with @wu-sheng about it more.

**Reply by kezhenxu94** [reply]
The problem to me is only that you missed the following part, which serves as a legend in the original graph and is missing in new design


![image](https://user-images.githubusercontent.com/15965696/133704002-62552180-ef81-496a-a735-c8b4a4299f64.jpeg)

**Reply by Superskyyy** [reply]
Would it be nice if we can find a set of four icons representing logging/metrics/tracing/events in a vivid way, then hovering the icons could bring up a tooltip saying it's whatever it is if people still cannot figure it out.

The reasoning is that current coloring - collection pair(green <-> logging) doesn't really form a meaningful connection, it is less ideal for newcomers to grab a memorable first impression.

Just a random thought. FYI my bro @rickyzhangca

**Reply by rickyzhangca** [reply]
I thought the proximity and style similarities are good enough for people to figure out what's going on but seems that it's not intuitive. Icons is hard as~17 icons will be clustering here. I'll spend some more time on this.
![image](https://user-images.githubusercontent.com/16908811/133944458-c5887d79-4f15-47db-b04b-9f1b1d701264.png)

**Reply by rickyzhangca** [reply]
better...? 

![image](https://user-images.githubusercontent.com/16908811/134772834-c522f1b6-c4ad-4928-a1fc-1590b7c5a1ed.png)

**Reply by wu-sheng** [reply]
How about we totally changed the style? Using the color of lines representing the trace/metric/log/event? And setting up sources from hardware to user application? Such as, from bottom to up, VM(Windows/Linux) -> K8s -> Mesh(Istio/Envoy) -> Backend Application(Multiple Language Agents) -> User End(Browser, maybe phone later). 
The key point of this graph to illuminate data flow(inbound and outbound), you don't have to be stuck by what I did.

**Reply by rickyzhangca** [reply]
! I might have totally misunderstood this graph from the beginning.

**Reply by rickyzhangca** [reply]
so...taking tracing for an example, the graph is actually saying we can `collect tracing data from front-end application/ backend application/service mesh and push it to transport layer`?
![image](https://user-images.githubusercontent.com/16908811/134777403-16b56796-6730-467a-8a5a-0ca0c6e5faf9.png)

**Reply by wu-sheng** [reply]
Yes, you can see this thing in 2 ways.
1. From catalog, tracing, metrics, logs, or events.
2. From source layer, OS, Container, Mesh, App, Service.

**Reply by rickyzhangca** [reply]
some explorations, the lines are not as clean as I wish them to be, hmm...

https://user-images.githubusercontent.com/16908811/134848454-d428c478-327e-4f76-8870-8e4b574c6a65.mov

![image](https://user-images.githubusercontent.com/16908811/134848459-de8ff165-6dbd-4b95-bc58-289390284782.png)

**Reply by wu-sheng** [reply]
This flow is better, could hou try to make the right part following this style too? The source could hold 50% of the whole page.

**Reply by rickyzhangca** [reply]
some tweaks before the other changes

https://user-images.githubusercontent.com/16908811/135020462-d0d9d66a-103a-4779-b53e-c141a6543a1b.mov

**Reply by wu-sheng** [reply]
This is cool. Let's follow this path. And we may need to update this graph from time to time. Hope we could have a good way.

**Reply by rickyzhangca** [reply]
More clean-ups! Maintainability: several approaches I am considering in my mind, should not be a problem if just adding/removing logos. Changing the paths and so on may require some more work but can be handled without a designer. 

https://user-images.githubusercontent.com/16908811/135124139-a0059aed-c6fc-43fa-b24a-2a34b53af2a2.mov

**Reply by wu-sheng** [reply]
The left side is nearly perfect. I like it. Just feel the right side is different style, such as rectangle VS rhombus, broken line VS oblique line. Also, icon style, could you try to make it clear? Such as, we have more information, Kafka/gRPC as transport layer implementation, different database as storage, etc.

**Reply by rickyzhangca** [reply]
how about:
![image](https://user-images.githubusercontent.com/16908811/135203279-22628b74-1ab0-4a55-acbd-095071b03c91.png)

**Reply by wu-sheng** [reply]
It is better. 
1. Let's merge receiver and aggregator but use a cube or something to present SkyWalking OAP server cluster?
2. Alarm should have several following notification channels, typically, Slack, PagerDuty, and Dingding
3. CLI and GUI should be replaced by the console and browser graphics?
4. The direction of CLI to SkyWalking arrow is opposite

**Reply by kezhenxu94** [reply]
> 4. The direction of CLI to SkyWalking arrow is opposite

If the arrow means the dataflow (seems to be true for other places), I think it's right. 

Also please note there is data from CLI to OAP and vice versa, CLI can report event data to OAP and retrieve data from OAP

**Reply by rickyzhangca** [reply]
1. Is `OAP server` a very skywalking vocabulary? OAP itself is generic but it seems that skywalking uses it the most (did a quick google). If so, considering the main audience of the landing page, it could be less clear than just `receiver cluster` and `aggregator cluster`. Though I doubt if people use `aggregator cluster` often either. Just my thought, updated the graphics. It does gives more room to breathe.
2.  Added! Technically should be Webhooks and gRPCHook? But I agree that showing the apps aligns with our goal better.
3. My thought is that the we only use graphics for the services supported to keep the graph clean. That's the idea behind many design decisions we've made so far and I think it works. Besides, using generic icons is inconsistent and `GUI/CLI` is more readable as well.
4. Good catch. I feel it's almost bi-directional? It can read and write. (edit: just saw @kezhenxu94 confirmed that)

![image](https://user-images.githubusercontent.com/16908811/135277993-ba188636-f292-4f79-8018-8795be17b921.png)

**Reply by wu-sheng** [reply]
1. Yes, putting a general name makes more sense.
2. Webhook is a general web Restful/JSON request, but slack/pagerduty are vendor specific.
3. Make sense.
4. Yes, it is bi-directional.

This is super better. Just feel SkyWalking cluster is a little thin, just feeling.

**Reply by rickyzhangca** [reply]
Thank you! Just made it taller. Probably at a good spot now before it becomes chunky. Feature locking this design to wrap it up for final hand-off.

![image](https://user-images.githubusercontent.com/16908811/135285803-09bbba68-646a-4869-9889-4cf84691acd3.png)

**Reply by Superskyyy** [reply]
I love it, looks so cool.

**Comment by wankai123** [comment]
Nice design!
I have one question about the content: should we put this on the home page?  
![image](https://user-images.githubusercontent.com/16773043/133580750-efe9bda5-e258-4b82-a62d-b481e6c29146.png)

**Reply by wu-sheng** [reply]
Once there is no specific project names or vendors, we are good.

**Reply by rickyzhangca** [reply]
Thanks! I hope it's a step up from the current site and will last long. 

Design wise, this section aims to provide detailed highlights to best sell SkyWalking, so user doesn't have to leave the site to google or spend time searching through our blogs by themselves. The comparison with competing/similar products are almost always one thing user will look for.

**Reply by rickyzhangca** [reply]
Policy wise, thank you wu-sheng for double checking!

**Comment by Jtrust** [comment]
It looks fantastic!
But I have a few questions: 
- What is the interaction of SkyWalking UI screenshots like on mobile? It looks like the click area is going to be really small.
![image](https://user-images.githubusercontent.com/21056967/133606997-f73c2b42-a7b1-4ecc-bf8b-91148dd204c8.png)

- Based on my personal preference, is it possible to change the rectangular border to other form? As I see, the rest of the page is mostly rounded design language.
![image](https://user-images.githubusercontent.com/21056967/133607556-fdc92c72-df6c-4f3c-82de-66f779253c7f.png)
![image](https://user-images.githubusercontent.com/21056967/133607999-94cd3096-0e74-4da7-9db2-dc26641f866c.png)

**Reply by rickyzhangca** [reply]
Screenshots section: 3 approaches (currently). The buttons won't be tiny. We can just switch to another set of graphics made for phones.

- Use screenshots from SkyWalking UI adapted to phone screen and include interactions. However we don't have the very best responsive design on the UI yet. Not sure if we are confident to show case it. E.g. Vercel does something similar.
- Keep using desktop screenshots but get rid of the interactions. Screenshots are horizontally scrollable. Not a common practice. 
- Simplify the entire section. E.g. Stripe does something similar.

What do you think?

![image](https://user-images.githubusercontent.com/16908811/133632179-b75800b9-5cba-44bf-af41-07fadfc807b8.png)

**Reply by rickyzhangca** [reply]
Round corners: good point! I started with some Apple-ish round corners there but I realized it's too much for us so I removed them. Glad to hear that we love some gentle round corners. Good balance between the sharp ones and Apple ones. We unintentionally did an AB testing here.

Updated the screens.

**Reply by Jtrust** [reply]
I prefer the design of the third one.

**Reply by Jtrust** [reply]
As far as I know, the width of the main display area on a general page is around 1200px. In this case, it's 1440px. Will it look too full? @rickyzhangca

**Comment by rickyzhangca** [comment]
@wu-sheng would we need to include the TM in our logo, if that's our "official logo"? I don't think ours in the [Apache Project logos](http://www.apache.org/logos/#skywalking) has one https://www.apache.org/foundation/marks/pmcs#graphics

**Reply by wu-sheng** [reply]
Most logos are submitted by some end users, which should be but missed from time to time.

**Reply by rickyzhangca** [reply]
ok I'll stick to the ones submitted to Apache for now

**Comment by Superskyyy** [comment]
Hey Ricky, great work! :heart_eyes:

**Comment by wu-sheng** [comment]
Noticed 2 things.
1. The directions of arrows in the architecture graph are not accurate. Most of them are one-way only. Also, the latest version seems too thin in the middle, but heavy on both sides.
2. Demo env requires user name and password to access, please provide a way to show them.

**Reply by rickyzhangca** [reply]
My bad! forgot to double check them after copy & paste. Fixed.

**Reply by rickyzhangca** [reply]
![image](https://user-images.githubusercontent.com/16908811/133914187-9784e96c-99e8-4bf9-83bf-b8c0d2f1a4da.png)

**Comment by wu-sheng** [comment]
Homepage is implemented through https://github.com/apache/skywalking-website/pull/352. @Jtrust leads.
